qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -112,7 +112,7 @@ def __init__(self, data, coeffs=None, *, ignore_pauli_phase=False, copy=True):
 
         pauli_list = PauliList(data.copy() if copy and hasattr(data, "copy") else data)
 
-        dtype = coeffs.dtype if isinstance(coeffs, np.ndarray) else complex
+        dtype = object if isinstance(coeffs, np.ndarray) and coeffs.dtype == object else complex
```

Comment:
LGTM, just one question: isn't `coeffs` always a `np.array`? At least that's what the type hint says 😄

Reply:
ok, thanks for the info! 👍🏻